### PR TITLE
Recipe for Perl module Encode

### PR DESCRIPTION
**Pull request by xileF1337**
This is a recipe for the Perl module Encode.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xileF1337** [comment]
@conda-forge/help-perl, ready for review!

File: recipes/perl-encode/gpl-1.0.txt [hunk 1]
```diff
@@ -0,0 +1,251 @@
+
```
Comment (cbrueffer) [inline recipes/perl-encode/gpl-1.0.txt:1]: Can be removed, see below.

File: recipes/perl-encode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{% set name = "perl-encode" %}
+{% set version = "3.02" %}
+{% set sha256 = "5865be4951870f62f43722818d076d7724306c75c8f268346b282351cbc820a8" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: Encode-{{ version }}.tar.gz
```
Comment (cbrueffer) [inline recipes/perl-encode/meta.yaml:10]: Can be removed I think?

File: recipes/perl-encode/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+test:
+  imports:
+    - Encode
+  commands:
+    - perl -we 'use Encode::Alias'  # version does not match that of Encode
+
+about:
+  home: http://metacpan.org/pod/Encode
+  license:  GPL-1.0-or-later
+  license_file: gpl-1.0.txt
```
Comment (cbrueffer) [inline recipes/perl-encode/meta.yaml:46]:
```suggestion
  license_file:
    - {{ environ["PREFIX"] }}/man/man1/perlartistic.1
    - {{ environ["PREFIX"] }}/man/man1/perlgpl.1
```

This is the pattern I've been using in other Perl recipes I've migrated from Bioconda; similar to R we can use license files that Perl already installs.

File: recipes/perl-encode/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+test:
+  imports:
+    - Encode
+  commands:
+    - perl -we 'use Encode::Alias'  # version does not match that of Encode
+
+about:
+  home: http://metacpan.org/pod/Encode
+  license:  GPL-1.0-or-later
```
Comment (cbrueffer) [inline recipes/perl-encode/meta.yaml:45]:
```suggestion
  license:  GPL-1.0-or-later OR Artistic-1.0-Perl
```

File: recipes/perl-encode/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  license_file: gpl-1.0.txt
+  summary: 'character encodings in Perl'
+
+extra:
+  recipe-maintainers:
+    - xileF1337
```
Comment (cbrueffer) [inline recipes/perl-encode/meta.yaml:51]: Feel free to add me as well.

File: recipes/perl-encode/build.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+# If it has Build.PL use that, otherwise use Makefile.PL
+if [ -f Build.PL ]; then
+    perl Build.PL
+    perl ./Build
+    perl ./Build test
+    # Make sure this goes in site
+    perl ./Build install --installdirs site
+elif [ -f Makefile.PL ]; then
+    # Make sure this goes in site
+    perl Makefile.PL INSTALLDIRS=site
```
Comment (cbrueffer) [inline recipes/perl-encode/build.sh:12]: I'm unsure about this. The proposed conda-forge Perl documentation at https://github.com/conda-forge/conda-forge.github.io/pull/1790 suggests `INSTALLDIRS=vendor`, and that's what I've been using for all recipes I've migrated. What do you think?

File: recipes/perl-encode/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  build:
+    - {{ compiler('c') }}
+
+  host:
+    - perl
+    - perl-storable
+    - perl-extutils-makemaker
+    - perl-exporter
+    - perl-parent
+    - make
```
Comment (cbrueffer) [inline recipes/perl-encode/meta.yaml:28]: make can be moved to the `build` section.

**Review by cbrueffer — COMMENTED** [review]
Looks good; I've added a dew comments/suggestions. I started migrating Perl recipes from Bioconda to conda-forge a short while ago, so feel free to tag in case you submit more Perl recipes.

File: recipes/perl-encode/build.sh (hunk 6 above)
Comment (xileF1337) [reply recipes/perl-encode/build.sh:12]: To be honest, I don't know Conda well enough to judge which paths are most appropriate. If you consider `vendor` to be the better option, I will use that. I took these files unchanged from the `conda skeleton cpan` output.

File: recipes/perl-encode/meta.yaml (hunk 2 above)
Comment (xileF1337) [reply recipes/perl-encode/meta.yaml:10]: Will do.

File: recipes/perl-encode/meta.yaml (hunk 5 above)
Comment (xileF1337) [reply recipes/perl-encode/meta.yaml:51]: Will do. If you wish I can also add you to future recipe submissions.

**Comment by xileF1337** [comment]
Is it normal for the linter to fail on subsequent commits ... ?!

**Comment by xileF1337** [comment]
@cbrueffer I incorporated your suggestions, could approve pls? Thanks!

**Comment by cbrueffer** [comment]
I just realized this is for Encode 3.02, which is quite old. The current version is 3.19. Is that on purpose?

**Comment by xileF1337** [comment]
> ?

The version was selected by `conda skeleton`, I don't know what's the logic behind its algorithm. I thought there may be a reason and decided to leave it, but I can update it if that's a bug.

**Comment by xileF1337** [comment]
@cbrueffer here we go, recipe updated to Encode 3.19.

**Pull request merged**